### DISCUSSION TITLE
[Question]: How to implement free-dall-e-proxy into librechat

### DISCUSSION THREAD

**Discussion by delta-whiplash** [danny-avila/LibreChat#2043] [Q&A] [answered]
### What is your question?

I just found this project : 
https://github.com/Feiyuyu0503/free-dall-e-proxy

And I wished to implement it in my actual librechat instance but I don't know where to start

### More Details

can someone help me to do this 😁

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment, marked as answer]
@delta-whiplash 

1. **Initial Setup:**
   Begin by seting up the free Dall-E proxy. Follow their instructions provided to get it operational. (For this you're on your own)

2. **Testing the Proxy:**
   After setting up the reverse proxy on your PC, verify its functionality by accessing the webUI at `http://localhost:8000/gradio/`. Successful image generation indicates proper setup.

3. **Configuration with LibreChat:**
   Integrate the proxy with LibreChat by updating the `.env` file with these variables:
   ```yaml
   DALLE_API_KEY=free-dall-e-proxy
   DALLE_REVERSE_PROXY=http://localhost:8000/v1/images/generations
   ```
**_Important Notes:_**

- **DALL-E-3 Plugin:** If previously installed in LibreChat, uninstall and then reinstall it.
- **Key and Port:** Ensure the free-dall-e-proxy key and port match your LibreChat configuration if defaults were altered.
- **OpenAI Endpoint:** For the “Plugins” endpoint functionality (invoking the Dall-E 3 plugin), an OpenAI compatible endpoint is required. Use your OpenAI API access or specify a reverse proxy address in `OPENAI_REVERSE_PROXY=` within the LibreChat `.env` file.

![image](https://github.com/danny-avila/LibreChat/assets/32828263/27ab6363-1bbb-4d8d-9fbc-ab4e4023831f)

**Reply by tensorstart** [reply]
@fuegovic hello,I have a similar problem
![QQMail_1710226519995.jpg](https://github.com/danny-avila/LibreChat/assets/40855931/e3668b74-c7de-4210-9b62-1acb106b710d)

I cant load pictures in chat window
Should I set up a reverse proxy
